src/OpenTK.Windowing.GraphicsLibraryFramework/JoystickState.cs

```diff
@@ -0,0 +1,235 @@
+//
+// JoystickState.cs
+//
+// Copyright (C) OpenTK
+//
+// This software may be modified and distributed under the terms
+// of the MIT license. See the LICENSE file for details.
+
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using OpenTK.Windowing.Common.Input;
+
+[assembly: InternalsVisibleTo("OpenTK.Windowing.Desktop")]
+
+namespace OpenTK.Windowing.GraphicsLibraryFramework
+{
+    /// <inheritdoc />
+    public sealed class JoystickState : IJoystickState
+    {
+        private const int MaxHistory = 2;
```

Comment:
Doc explaining what this means would be good.

If we're going with a fancy generalized implementation it would be much better to contain it in a proper data structure `RingBuffer<T>` instead of handling that logic here.

I'd generally suggest against that though and going with two simple fixed fields will be way simpler.

Reply:
All this really does is switch the current and last between the two indexes. I can remove allusions to it being a ring buffer / circular buffer, it really is just two arrays.

I prefer this approach because it means we never have to Array.Copy the values from current state to previous state, we just swap them and overwrite the new current with new values.